File: backend/ms-users/src/main/java/pi/ms_users/domain/Notice.java
package pi.ms_users.domain;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Notice")
public class Notice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private String userId;

    @Column(name = "date", nullable = false)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private LocalDateTime date;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "description", nullable = false)
    private String description;
}

File: frontend/src/app/user/types/notice.ts
export interface Notice {
  id: number;
  userId: string;
  date: string;
  title: string;
  description: string;
  mainImage: string | File | null;
}

export type NoticeCreate = {
  userId: string;
  title: string;
  description: string;
  mainImage: File;
};